zephyr/include/rtos/cache.h

```diff
@@ -6,14 +6,72 @@
 #ifndef __ZEPHYR_RTOS_CACHE_H__
 #define __ZEPHYR_RTOS_CACHE_H__
 
-/* TODO: align with Zephyr generic cache API when ready */
 #define __SOF_LIB_CACHE_H__
-#include <arch/lib/cache.h>
+
+#if !defined(__ASSEMBLER__) && !defined(LINKER)
+
+#include <zephyr/cache.h>
+#include <zephyr/debug/sparse.h>
+
+#if defined(CONFIG_XTENSA) && defined(CONFIG_INTEL)
+
+/* definitions required by xtensa-based Intel platforms.
+ *
+ * TODO: if possible, move these to Zephyr.
+ */
+#define SRAM_UNCACHED_ALIAS 0x20000000
+#define is_cached(address) (!!((uintptr_t)(address) & SRAM_UNCACHED_ALIAS))
+
+#endif /* defined(CONFIG_XTENSA) && defined(CONFIG_INTEL) */
 
 /* writeback and invalidate data */
 #define CACHE_WRITEBACK_INV	0
 
 /* invalidate data */
 #define CACHE_INVALIDATE	1
 
+#if !defined(CONFIG_DCACHE_LINE_SIZE_DETECT) && \
+	(CONFIG_DCACHE_LINE_SIZE > 0)
+
+#define DCACHE_LINE_SIZE CONFIG_DCACHE_LINE_SIZE
+
+#else
+
+/* If CONFIG_DCACHE_LINE_SIZE is not set appropriately then
+ * fallback to a default value. This ought to be alright
+ * since all platforms that support Zephyr use this value.
+ * This is better than forcing all vendors to set
+ * CONFIG_DCACHE_LINE_SIZE in order to avoid compilation
+ * errors.
+ */
+#define DCACHE_LINE_SIZE 128
```

Comment:
I think it's 64 at least on Intel platforms

Reply:
yup, rn I'm working on adding a new configuration (some sort of wrapper over CONFIG_DCACHE_LINE_SIZE from Zephyr) which will be the equivalent of all of those messy ifdefs